dali/pipeline/operators/util/external_source.h

```diff
@@ -18,29 +18,110 @@
 #include <atomic>
 #include <string>
 #include <vector>
-#include <condition_variable>
+#include <memory>
+#include <list>
 #include <mutex>
+#include <condition_variable>
+#include <utility>
 
 #include "dali/pipeline/operators/operator.h"
+#include "dali/pipeline/util/worker_thread.h"
 
 namespace dali {
+namespace detail {
+
+struct CudaEventWrapper {
+  CudaEventWrapper() {
+    auto res = cudaEventCreate(&event);
+    if (res != cudaSuccess) {
+      DALI_WARN("Fatal error: " + to_string(res) + " in CudaEventWrapper()");
```

Comment:
warning and terminate? Wondering, why not use DALI_ENFORCE?

Reply:
DALI_ENFORCE throws, while we want explicitly terminate.